Automattic/wp-calypso
# Jetpack Cloud: Get scan state for the selected site from the API

#### Changes proposed in this Pull Request

* Adds `getJetpackScanState` selector
* Uses `getJetpackScanState` selector to get scan state from the API, instead of from the URL query string
* Shows correct state in the UI (e.g., "scanning," "threat," "okay," "error")
* Adds loading state to Scan page while state from API is still pending
* Moves `<QueryJetpackScan/>` component from the Threats page to the main Scan page
* Adds missing function dependencies to `useEffect` in QueryJetpackScan component

#### Testing instructions



* Load Jetpack Cloud for any site. Ensure that you see the pulsing "loading" placeholder on the main Scan page before the scan state is shown.
* Load Jetpack Cloud for a site with threats. Ensure that threats are shown on the main Scan page.
* Load Jetpack Cloud for a site with no threats. Ensure that the "okay" state is shown on the main Scan page.

#### Screenshots

##### Loading state (with placeholder)

<img width="1064" alt="Loading state" src="https://user-images.githubusercontent.com/670067/79386447-c267e280-7f2f-11ea-873a-e94e0ccaddec.png">

##### Threats state

<img width="1041" alt="Threats state" src="https://user-images.githubusercontent.com/670067/79386793-48842900-7f30-11ea-9a69-f9d45db6c199.png">

Fixes 1168549268277608-as-1170788733452764
